rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -312,20 +317,20 @@ public RMCore(String id, Node nodeRM) throws ActiveObjectCreationException, Node
         this.id = id;
         this.nodeRM = nodeRM;
 
-        nodeSources = new HashMap<>();
-        brokenNodeSources = new ArrayList<>();
+        deployedNodeSources = new HashMap<>();
+        definedNodeSources = new HashMap<>();
         allNodes = new ConcurrentHashMap<>();
         eligibleNodes = Collections.synchronizedList(new ArrayList<RMNode>());
 
         this.accountsManager = new RMAccountsManager();
         this.jmxHelper = new RMJMXHelper(this.accountsManager);
     }
 
-    public RMCore(Map<String, NodeSource> nodeSources, List<String> brokenNodeSources, Map<String, RMNode> allNodes,
-            Client caller, RMMonitoringImpl monitoring, SelectionManager manager, List<RMNode> freeNodesList,
-            RMDBManager newDataBaseManager) {
-        this.nodeSources = nodeSources;
-        this.brokenNodeSources = brokenNodeSources;
+    public RMCore(Map<String, NodeSource> deployedNodeSources, Map<String, NodeSourceDescriptor> definedNodeSources,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)